### PR TITLE
chore: Fix internal tslint errors

### PR DESCRIPTION
**Pull request by acdvorak**
Refs #4225

### CODE REVIEW

File: packages/mdc-dom/ponyfill.ts [hunk 1]
```diff
@@ -26,7 +26,6 @@
  * This makes ponyfills safer than traditional polyfills, especially for libraries like MDC.
  */
 
-export function closest<T extends Element>(element: Element, selector: string): T | null;
```
[the remark this answers is not in the document]
Comment (moog16) [reply packages/mdc-dom/ponyfill.ts]: I guess we're not using the generic if all tests pass, but is this something that google3 is throwing a warning about? What is the warning/error?

File: packages/mdc-base/component.ts [hunk 2]
```diff
@@ -118,4 +118,6 @@ export class MDCComponent<FoundationType extends MDCFoundation> {
   }
 }
 
+// tslint:disable:no-default-export Needed for backward compatibility with
+// MDC Web v0.44.0 and earlier.
```
Comment (moog16) [inline packages/mdc-base/component.ts]: we don't need this in the other components/foundations/adapters? Does this pass if we revert to:

```
export {MDCComponent, MDCComponent as default}
```

File: packages/mdc-dom/ponyfill.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (acdvorak) [reply packages/mdc-dom/ponyfill.ts]: The internal linter complained that this isn't a useful type constraint and won't do what callers expect.

They specifically referenced this _exact_ use case (`<T extends Element>(): T`) to illustrate their point lol.

Instead, the caller is supposed to explicitly cast the returned value to the desired type.

Also, as Matt mentioned, we're not using the generic anywhere anyway.

File: packages/mdc-tab-bar/foundation.ts [hunk 3]
```diff
@@ -26,7 +26,11 @@ import {MDCTabDimensions, MDCTabInteractionEvent} from '@material/tab/types';
 import {MDCTabBarAdapter} from './adapter';
 import {numbers, strings} from './constants';
 
+// tslint:disable:no-inferrable-new-expression Keep explicit generic param for
```
[the remark this answers is not in the document]
Comment (acdvorak) [reply packages/mdc-tab-bar/foundation.ts]: I rewrote them as `new Set<string>()` and `new Map<number, string>()` instead.

Good call!

File: packages/mdc-base/component.ts (hunk 2 above)
Comment (acdvorak) [reply packages/mdc-base/component.ts]: It still complains with `export {MDCComponent, MDCComponent as default}`.

In `master`, the only files that have default exports are all the foundations,
`mdc-auto-init/index.js`, and `mdc-base/component.js`:

https://github.com/material-components/material-components-web/blob/70672f12ff19591c53fef6876c059c854f9cb816/packages/mdc-base/component.js#L131

**Pull request merged**